cranelift/codegen/src/machinst/isle.rs

```diff
@@ -19,6 +20,8 @@ pub type OptionWritableReg = Option<WritableReg>;
 pub type VecReg = Vec<Reg>;
 pub type VecWritableReg = Vec<WritableReg>;
 pub type ValueRegs = crate::machinst::ValueRegs<Reg>;
+pub type InstOutput = Vec<ValueRegs>;
```

Comment:
On general comment -- we should look to make sure in whatever final design we use, we don't incur any allocations in the ordinary one-register case. We use `SmallVec` in a number of other places in the backend framework, and some of the chosen inline-vector sizes were based on allocation profiling; we found sometimes significant speedups from this at least early in development. I think we can definitely get around this though (see top-level comment).